dashboard/factories.py

```diff
@@ -47,3 +50,27 @@ class CachedEnrollmentFactory(DjangoModelFactory):
 
     class Meta:  # pylint: disable=missing-docstring,no-init,too-few-public-methods,old-style-class
         model = CachedEnrollment
+
+
+class ProgramEnrollmentFactory(DjangoModelFactory):
+    """Factory for ProgramEnrollment"""
+    user = SubFactory(UserFactory)
+    program = SubFactory(ProgramFactory)
+
+    class Meta:  # pylint: disable=missing-docstring,no-init,too-few-public-methods,old-style-class
+        model = ProgramEnrollment
+
+    @classmethod
+    def create(cls, **kwargs):
+        # ProgramEnrollments should only exist if there is a CachedEnrollment associated with
+        # the given User and Program. Instead of creating a new record with the factory, we
+        # will create the necessary objects to trigger its creation.
+        user = kwargs['user'] = kwargs.get('user', UserFactory.create())
+        program = kwargs['program'] = kwargs.get('program', ProgramFactory.create())
+        course = CourseFactory.create(program=program)
+        course_run = CourseRunFactory.create(course=course)
+        CachedEnrollmentFactory.create(user=user, course_run=course_run)
+        CachedCertificateFactory.create(user=user, course_run=course_run)
```

Comment:
technically no, but it seems that in other test cases, we prefer to have an enrollment and a certificate. i put this in for simple convenience